java/arcs/android/common/TraceUtils.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.android.common
+
+import android.os.Trace
+
+/**
+ * Harness [android.os.Trace] trace points identified by
+ * the trace [tag] to the entry and the exit of the [block].
+ */
+inline fun scopedTrace(tag: String, block: () -> Unit) {
```

Comment:
This might be more flexible going forward if it were a generic function that accepts a generic-returning block:

```kotlin
inline fun <T> scopedTrace(tag: String, block: () -> T): T {
    Trace.beginSection(tag)
    val result = block()
    Trace.endSection()
    return result
}
```

Reply:
Also, is there much overhead in the Trace calls?

Reply:
Good suggestion! Thanks.

re overhead, 
when tracing is disabled (default), the overhead of calling Trace apis is roughly equivalent to read a tag variable in memory and check if corresponding tracing flag is set. So the overhead can be neglected.

when tracing is enabled, the overhead is that in above plus writing a trace point to per-cpu ring buffer, which usually takes single-digit micro-seconds.

Reply:
done